### PR TITLE
Exclude units without spikes from all calculations (still a bug here)

### PR DESCRIPTION
**Pull request by mhhennig**
This code still breaks when `unit_ids` is supplied. The trouble is that the metric_data is computed for all units in an extractor, regardless of `unit_ids`.

### CODE REVIEW

**Comment by colehurwitz** [comment]
How I will fix this is to make the `unit_ids = sorting.get_unit_ids()` only when `unit_ids == None`. When unit_ids is a list of unit ids, I would do:  `unit_ids = intersection(unit_ids, sorting.get_unit_ids())`

File: spiketoolkit/validation/metric_calculator.py [hunk 1]
```diff
@@ -42,6 +43,15 @@ def __init__(self, sorting, recording=None, sampling_frequency=None, unit_ids=No
...
             unit_ids = sorting.get_unit_ids()
+            
+        # only use units with spikes to avoid breaking metric calculation
+        num_spikes_per_unit = [len(sorting.get_unit_spike_train(u)) for u in unit_ids]
+        sorting = ThresholdCurator(sorting=sorting, metrics_epoch=num_spikes_per_unit)
+        sorting.threshold_sorting(0, 'less_or_equal')
+
+        unit_ids = sorting.get_unit_ids()
+        if len(unit_ids)==0:
+            raise ValueError("No spikes found.")
```
Comment (samuelgarcia) [inline spiketoolkit/validation/metric_calculator.py]: I would prefer "no units found"

Comment (colehurwitz) [reply spiketoolkit/validation/metric_calculator.py]: Fixed in new commits

**Comment by colehurwitz** [comment]
Ready to merge

**Pull request merged**